### PR TITLE
✨ Add metadata only to template language

### LINKED ISSUE #227: Placeholders metadata only in template language
It would be great if placeholders metadata was added only to the template language since it's not necessary elsewhere.

Either way I really like this plugin

### PR DESCRIPTION
**Pull request by RobertOdrowaz**
Resolves #227 

FYI I wasn't able to run the tests because of some weird error
```
[vscode.php]: One or more snippets from the extension 'php' very likely confuse snippet-variables and snippet-placeholders (see https://code.visualstudio.com/docs/editor/userdefinedsnippets#_snippet-syntax for more details)
Unexpected token '}', "}^��qV6$"... is not valid JSON
Error: Cannot find module '/home/robert/Documents/l10nization/out/test/suite/index'
```

### CODE REVIEW

**Comment by lsaudon** [comment]
Wow, thanks for your contribution.
I'll try to look at that later today.

**Comment by lsaudon** [comment]
Thank you very much, I didn't know that wasn't necessary.
I don't think I'm the only one.
I assume that's also the case for the description?

End-to-end tests are broken.
I'm only running unit tests at the moment 😞  `npm run unit-test`

File: src/test/suite/toJson.spec.ts [hunk 1]
```diff
@@ -442,6 +460,40 @@ describe('toJson', () => {
...
+  "@@locale": "fr",
+  "helloWorld": "Hello World"
+}`);
+  });
+
+  it('should return json when message with 1 placeholder for non template file', () => {
+    expect(
+      toJson(
+        defaultArbJson,
+        true,
```
Comment (lsaudon) [inline src/test/suite/toJson.spec.ts:487]: replace `true` to `false`

File: src/test/suite/toJson.spec.ts [hunk 2]
```diff
@@ -442,6 +460,40 @@ describe('toJson', () => {
...
     }
   }
+}`);
+  });
+
+  it('should return json when message with description non tempate file', () => {
+    expect(
+      toJson(
+        defaultArbJson,
+        true,
```
Comment (lsaudon) [inline src/test/suite/toJson.spec.ts:470]: replace `true` to `false`

File: src/test/suite/toJson.spec.ts [hunk 3]
```diff
@@ -10,21 +10,24 @@ const defaultArbJson = `{
...
+    expect(
+      toJson(defaultArbJson, true, 'helloWorld', null, 'Hello World', [], false)
+    ).to.be.equal(`{
   "@@locale": "fr",
   "helloWorld": "Hello World"
 }`);
   });
 
-  it('should return json when message with description', () => {
+  it('should return json when message with description for tempate file', () => {
```
Comment (lsaudon) [inline src/test/suite/toJson.spec.ts:21]: `template` to `template`

**Review by lsaudon — CHANGES_REQUESTED** [review]

**Comment by RobertOdrowaz** [comment]
I've fixed the tests.

I don't think that placeholders and description only in template file is explicitly stated anywhere in the documentation. It can be found in the generator code itself https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/localizations/gen_l10n_types.dart#L343
```dart
description = _description(templateBundle.resources, resourceId, isResourceAttributeRequired),
placeholders = _placeholders(templateBundle.resources, resourceId, isResourceAttributeRequired),
```
both description and placeholders take only `templateBundle` argument.

As a sidenote there is a comment above `Message` class is incorrect stating that every entry should have corresponding @ entry in the template file which clearly isn't true.
```dart
// The template ARB file must contain an entry called @myResourceId for each
// message named myResourceId. The @ entry describes message parameters
// called "placeholders" and can include an optional description.
// Here's a simple example message with no parameters:
```

**Review by lsaudon — APPROVED** [review]

**Pull request merged**